### PR TITLE
Ins 25712 implement api endpoints to manage all firewall rules of a cluster

### PR DESCRIPTION
**Pull request by ankit-agrawal26**

### CODE REVIEW

File: docs/resources/aws_security_group_firewall_rules_v2.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+---
+page_title: "instaclustr_aws_security_group_firewall_rules_v2 Resource - terraform-provider-instaclustr"
+subcategory: ""
+description: |-
+---
+
+# instaclustr_aws_security_group_firewall_rules_v2 (Resource)
+A listable data source of all AWS Security Group based firewall rules in an Instaclustr managed cluster.
```
Comment (carlsandland) [inline docs/resources/aws_security_group_firewall_rules_v2.md:8]: this is a resource not a datasource?

File: docs/resources/cluster_network_firewall_rules_v2.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
+---
+page_title: "instaclustr_cluster_network_firewall_rules_v2 Resource - terraform-provider-instaclustr"
+subcategory: ""
+description: |-
+---
+
+# instaclustr_cluster_network_firewall_rules_v2 (Resource)
+A listable data source of all CIDR based firewall rules in an Instaclustr managed cluster.
```
Comment (carlsandland) [inline docs/resources/cluster_network_firewall_rules_v2.md:8]: this is a resource not a datasource?

File: docs/resources/aws_security_group_firewall_rules_v2.md (hunk 1 above)
Comment (ankit-agrawal26) [reply docs/resources/aws_security_group_firewall_rules_v2.md:8]: The GET request(instaclustr_aws_security_group_firewall_rules_v2_instance) would be its datasource

**Review by carlsandland — APPROVED** [review]

File: docs/data-sources/aws_security_group_firewall_rules_v2_instance.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+*___deferred_reason___*<br>
+<ins>Type</ins>: string, read-only<br>
+<br>The reason (if needed) for the deferred status of the AWS security group firewall rule.<br><br>
+*___id___*<br>
+<ins>Type</ins>: string (uuid), read-only<br>
+<br>ID of the AWS security group firewall rule.<br><br>
+*___type___*<br>
+<ins>Type</ins>: string, read-only<br>
+<ins>Constraints</ins>: allowed values: [ `APACHE_ZOOKEEPER`, `CADENCE`, `CADENCE_GRPC`, `CADENCE_WEB`, `CASSANDRA`, `CASSANDRA_CQL`, `ELASTICSEARCH`, `KAFKA`, `KAFKA_CONNECT`, `KAFKA_ENCRYPTION`, `KAFKA_MTLS`, `KAFKA_NO_ENCRYPTION`, `KAFKA_PRIVATE_SASL_PLAINTEXT_LISTENER`, `KAFKA_PUBLIC_SASL_PLAINTEXT_LISTENER`, `KAFKA_PRIVATE_PLAINTEXT_LISTENER`, `KAFKA_PUBLIC_PLAINTEXT_LISTENER`, `KAFKA_PRIVATE_SSL_LISTENER`, `KAFKA_PUBLIC_SSL_LISTENER`, `KAFKA_REST_PROXY`, `KAFKA_SCHEMA_REGISTRY`, `KARAPACE_REST_PROXY`, `KARAPACE_SCHEMA_REGISTRY`, `MONGODB`, `OPENSEARCH`, `OPENSEARCH_DASHBOARDS`, `PGBOUNCER`, `POSTGRESQL`, `REDIS`, `SEARCH_DASHBOARDS`, `SECURE_APACHE_ZOOKEEPER`, `SPARK`, `SPARK_JOBSERVER`, `SHOTOVER_PROXY`, `DEBEZIUM_CONNECTOR_CASSANDRA_KAFKA`, `DEBEZIUM_CONNECTOR_CASSANDRA_SCHEMA` ]<br><br>The type of firewall rule.<br><br>
```
Comment (iclonghoang) [inline docs/data-sources/aws_security_group_firewall_rules_v2_instance.md:46]: I think you did not sync your branch with `releases` so this list is missing a recently added value
it's no biggie but next PRs would see unrelated changes in your new files here

Comment (ankit-agrawal26) [reply docs/data-sources/aws_security_group_firewall_rules_v2_instance.md]: Have updated it

Comment (iclonghoang) [reply docs/data-sources/aws_security_group_firewall_rules_v2_instance.md]: your version bump is gone

Comment (ankit-agrawal26) [reply docs/data-sources/aws_security_group_firewall_rules_v2_instance.md]: No problem, I will have to update it at the time of production, because anyone can come up between that :)

**Review by carlsandland — APPROVED** [review]

**Review by iclonghoang — APPROVED** [review]

**Review by carlsandland — APPROVED** [review]
